custom/ewsghana/forms.py

```diff
@@ -109,7 +109,7 @@ def __init__(self, *args, **kwargs):
         self.fields['facility'].widget = FacilitiesSelectWidget(domain=domain, id='facility')
 
     def save(self, user, domain):
-        ews_extension = EWSExtension.objects.get_or_create(user_id=user.get_id)[0]
+        ews_extension = EWSExtension.objects.get_or_create(user_id=user.get_id, domain=domain)[0]
```

Comment:
@kkrampa I'm not sure I understand how this fixes the issue - user.get_id is unique, so adding domain to it won't make it more unique. I think it's more likely that multiple extension objects are created because of a race condition.